src/ConstructingAggregateRootRepository.php

```diff
@@ -8,12 +8,12 @@
 use function assert;
 use function count;
 
-final class ConstructingAggregateRootRepository implements AggregateRootRepository
+class ConstructingAggregateRootRepository implements AggregateRootRepository
```

Comment:
Why don't you decorate this repo instead of extending it?

Reply:
That's one of the discussion points.

Reply:
Considering that the interfaces are more important, this change will happen so snapshotting is also available to those who don't use the default construction strategy.